crates/rattler_installs_packages/src/index/http.rs

```diff
@@ -370,3 +429,73 @@ fn body_to_streaming_or_local(
             .compat(),
     ))
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::index::{
+        file_store::FileStore,
+        http::{write_cache_bom, CACHE_BOM, CURRENT_VERSION},
+    };
+    use http::{header::CACHE_CONTROL, HeaderMap, HeaderValue, Method};
+    use reqwest::Client;
+    use reqwest_middleware::ClientWithMiddleware;
+
+    use std::{io::BufWriter, sync::Arc};
+    use tempfile::TempDir;
+
+    use super::{key_for_request, read_cache, CacheMode, Http};
+
+    fn get_http_client() -> (Arc<Http>, TempDir) {
+        let tempdir = tempfile::tempdir().unwrap();
+        let client = ClientWithMiddleware::from(Client::new());
+
+        let http = Http::new(
+            client,
+            FileStore::new(&tempdir.path().join("http")).unwrap(),
+        );
+
+        (Arc::new(http), tempdir)
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    pub async fn test_cache_is_correct_written_and_read_when_requesting_pypi_boltons() {
+        let url = url::Url::parse("https://pypi.org/simple/boltons").unwrap();
+
+        let url_clone = url.clone();
+
+        let (client_arc, _tmpdir) = get_http_client();
+
+        let mut headers = HeaderMap::new();
+        headers.insert(CACHE_CONTROL, HeaderValue::from_static("max-age=0"));
+
+        // let's make a request and validate that the cache is saved
+        client_arc
+            .request(url, Method::GET, headers.clone(), CacheMode::Default)
+            .await
+            .unwrap();
+
+        let key = key_for_request(&url_clone, Method::GET, &headers);
+        let lock = client_arc.http_cache.lock(&key.as_slice()).await.unwrap();
+
+        let res = lock.reader().and_then(|reader| {
+            read_cache(reader.detach_unlocked(), CACHE_BOM, CURRENT_VERSION).ok()
+        });
+
+        assert!(res.is_some());
+
+        let mut buf_writer = BufWriter::new(lock.begin().unwrap());
+
+        // let's "corrupt" cache and change it's bom predenting that it's older or different cache
```

Comment:
Again I would not call version part of the `bom`

Reply:
rename it